src/core/coap/coap_message.hpp

```diff
@@ -97,6 +97,16 @@ class Message : public ot::Message
      */
     typedef otCoapCode Code;
 
+    /**
+     * CoAP Block1/Block2 Types
+     *
+     */
+    typedef enum otCoapBlockType
```

Comment:
@nealjack thanks for all the changes. Some small naming suggestions below:

In OT, core is c++, but all public APIs as defined in C.

Since this definition is restricted to core (c++ source), we don't need the `typedef` anymore (`enum Type` already defines a type named `Type`).

Also the `ot{Name}` convention is typically used for public C OpenThread definitions. Within core modules, we don't use the `ot` prefix (since everything is already under `namespace ot`).

For constants, the `OT_{NAME}` is typically used for public C OT definitions. Within core (c++) the convention is `k{NameInCamelCase`}`.


So suggest the following:
```
enum BlockType  
{
    kBlockType1 = 1,
    kBlockType2 = 2,
};
```
(this `enum` type is already under `class Coap` so it would get namespaced as `Coap::BlockType`).

Reply:
Sounds good - I moved the enum definition without thinking about naming